### PR TITLE
Update Old Python versions to modern supported ones

### PR DESCRIPTION
**Pull request by clintonsteiner**
Python 3.8 is EOL - update all old references to newer versions

## Pre-requisites

- [x] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [x] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [x] Should include comments in the workflow for any parts that are not obvious or could use clarification.
- [x] Should specify least privileged [permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) for `GITHUB_TOKEN` so that the workflow runs successfully.

**For _CI_ workflows, the workflow:**

- [x] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [x] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [x] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [x] Packaging workflows should run on `release` with `types: [ created ]`.
- [x] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [x] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/code-scanning).
- [x] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [x] `name`: Name of the Code Scanning integration.
  - [x] `creator`: Name of the organization/user producing the Code Scanning integration.
  - [x] `description`: Short description of the Code Scanning integration.
  - [x] `categories`: Array of languages supported by the Code Scanning integration.
  - [x] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [x] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [x] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [x] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [x] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [x] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

File: code-scanning/mobsf.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
       - name: Setup python
         uses: actions/setup-python@v3
         with:
-          python-version: 3.8
+          python-version: 3.13
```
Comment (reactive-firewall) [inline code-scanning/mobsf.yml:33]: 😕 Question: why bump python version and not action version? Is there a reason?

Please see [setup-python action releases](https://github.com/actions/setup-python/releases) Note the changes for version `v5` includes migrating the underlying node version (which fixes a deprecated version just like the 3.8 to 3.13 does).

I recommend something like this:

```suggestion
        uses: actions/setup-python@v5
        with:
          python-version: "3.13"  # set this to the python version to use (see https://github.com/actions/setup-python?tab=readme-ov-file#basic-usage)
```

> [!CAUTION]
> Also please note that `python-versions` after `3.10` and non-`cpython` need quoting to properly function in some GitHub Action Runner Environments (in contrast with `3.8` which did not need quotes); compare with the format used in `ci/pylint.yml` where quotes are used.

### Benefits of these changes:

* **Syntacticly Correct**: Uses quotes around `python-version` correctly.
* **Consistency in Comments**: Helps keep comments consistent with other changes in this PR.
* **Improved Security**: Both the original version bump to 3.13 and the `setup-python` version bump to `v5` are important for remaining current with security updates.

File: ci/django.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
     strategy:
       max-parallel: 4
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.11, 3.12, 3.13]
```
Comment (reactive-firewall) [inline ci/django.yml:16]: 🤔 I'd recommend the following:

```suggestion
        python-version: ["3.11", "3.12", "3.13"]
```

> [!CAUTION]
> Also please note that `python-versions` after `3.10` and non-`cpython` need quoting to properly function in some GitHub Action Runner Environments (in contrast with `3.7`/`3.8`/`3.9` which do not need quotes); compare with the format used in `ci/pylint.yml` where quotes are used.

### Benefits of these changes:

* **Syntacticly Correct**: Uses quotes around `python-version` correctly.

---

> [!TIP]
> Please see [setup-python action releases](https://github.com/actions/setup-python/releases) Note the changes for version `v5` includes migrating the underlying node version (which fixes a deprecated version just like the 3.7-3.9 to 3.11-3.13 does).

File: ci/pylint.yml [hunk 3]
```diff
@@ -7,7 +7,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10"]
+        python-version: ["3.11", "3.12", "3.13"]
```
Comment (reactive-firewall) [inline ci/pylint.yml:10]: :+1: LGTM!

> [!TIP]
> Please see [setup-python action releases](https://github.com/actions/setup-python/releases) Note the changes for version `v5` includes migrating the underlying node version (which fixes a deprecated version just like the 3.8 to 3.11 change does).

**Review by reactive-firewall — CHANGES_REQUESTED** [review]

File: deployments/azure-webapps-python.yml [hunk 4]
```diff
@@ -21,7 +21,7 @@ name: Build and deploy Python app to Azure Web App
 
 env:
   AZURE_WEBAPP_NAME: your-app-name  # set this to the name of your Azure Web App
-  PYTHON_VERSION: '3.8'              # set this to the Python version to use
+  PYTHON_VERSION: '3.13'             # set this to the Python version to use (e.g. '3.11', '3.12', '3.13')
```
Comment (reactive-firewall) [inline deployments/azure-webapps-python.yml:24]: > [!NOTE]
> **TL;DR** This is progress, however the comment alignment issue still remains [when viewed as a file](https://github.com/actions/starter-workflows/blob/eeb2dc6f8c6d9cb81eb9a0b0b33beb71de7eaf97/deployments/azure-webapps-python.yml#L23-L24).

File: ci/django.yml (hunk 2 above)
Comment (clintonsteiner) [reply ci/django.yml:16]: Fixed

File: code-scanning/mobsf.yml (hunk 1 above)
Comment (clintonsteiner) [reply code-scanning/mobsf.yml:33]: fixed

**Comment by clintonsteiner** [comment]
Fixed all comments - asking reactive-firewall for rereview

**Review by reactive-firewall — COMMENTED** [review]
👍 LGTM.

**Review by mScott224 — APPROVED** [review]

**Review by reactive-firewall — APPROVED** [review]

**Comment by clintonsteiner** [comment]
Bumping pr

**Review by JamesonDiamond1 — APPROVED** [review]
@JamesonDiamond1
